files/pgbouncer_info.py

```diff
@@ -0,0 +1,130 @@
+from collections import defaultdict
+
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+import collectd
+import psycopg2
+
+# Default host, override in config by specifying 'Host'
+HOST = "localhost"
+
+# Default port, override in config by specifying 'Port'
+PORT = 5432
+
+# Default database, override in config by specifying 'Database'
+DBNAME = "pgbouncer"
+
+# Default user, override in config by specifying 'User'
+USER = "postgres"
+
+# Default password, override in config by specifying 'Password'
+PASSWORD = ""
+
+
+def get_stats():
+    conn, cur = None, None
+    stats = {}
+    try:
+        conn = psycopg2.connect(
+            database=DBNAME,
+            user=USER,
+            password=PASSWORD,
+            host=HOST,
+            port=PORT
+        )
+        conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+        cur = conn.cursor()
+        stats = _get_stats(cur)
+    finally:
+        if cur is not None:
+            cur.close()
+            if conn is not None:
+                conn.close()
+
+    return stats
+
+
+def _get_stats(cur):
+    stats = defaultdict(dict)
+
+    cur.execute("SHOW STATS;")
+    for (database, total, _, _, _, req, recv,
+         sent, query) in cur.fetchall():
+        stats[database] = {
+            'total_requests': total,
+            'req_per_sec': req,
+            'recv_per_sec': recv,
+            'sent_per_sec': sent,
+            'avg_query': query
+        }
+
+    cur.execute("SHOW POOLS;")
+    for (database, _, cl_active, cl_waiting, sv_active, sv_idle,
+         sv_used, sv_tested, sv_login, maxwait, _) in cur.fetchall():
+        values = {
+            'cl_active': cl_active,
+            'cl_waiting': cl_waiting,
+            'sv_active': sv_active,
+            'sv_idle': sv_idle,
+            'sv_used': sv_used,
+            'sv_tested': sv_tested,
+            'sv_login': sv_login,
+            'maxwait': maxwait
+        }
+        if database not in stats:
+            stats[database] = dict()
+        # there can be many lines for one database, one for each user
```

Comment:
This is not true.

Reply:
Okay. It was there in original script from the link.